<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd"><html lang="en-GB"><html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en-gb">
<head>
  <meta name="author" content="et3 (Electra Tapanari)">
  <meta name="template-svn-id" content="$Id$">

  <link rel="Shortcut Icon" href="/gencode/gfx/gencodegenes-logo-helix-av.png" type="image/png">
  <title>GENCODE - GENCODE Release Files</title>
  <link rel="stylesheet" type="text/css" href="/t/cssjs/l7wWtfKGenfNZMD2xQzoIQ.c.css">
<!--[if lt IE 9]>
  <link rel="stylesheet" type="text/css" href="/t/cssjs/1ElZ9XP7DepH4jLsZsoOqQ.c.css">
<![endif]-->
<!--[if lt IE 8]>
  <link rel="stylesheet" type="text/css" href="/t/cssjs/186Y_6KZXC9L3oWbvaBcIw.c.css">
<![endif]-->

  <meta name="svn-id" content="$Id: release_link_file.pl 326 2012-03-20 12:26:10Z jmg 19.html 87 2011-07-03 16:19:38Z jmg $">
</head>
<body>
  <div id="wrap">
    <div id="leftimg">
      <a href="/"><img src="/gencode/gfx/gencodegenes-logo-helix-av.png" alt="GENCODE" ></a><div style="float: right;"></div>
    </div>
    <div id='cssmenu'>
      <ul>
        <li class='has-sub'><a href='#'><span>GENCODE</span></a>
          <ul>
            <li><a href="/about.html">About</a></li>
            <li><a href="/custom_array.html">GENCODE Custom Array</a></li>
            <li><a href="/rgasp/">GENCODE Benchmarking</a></li>
            <li><a href="/publications.html">Publications</a></li>
            <li><a href="/outreach.html">GENCODE Outreach</a></li>
            <li><a href="/faq.html">FAQ</a></li>
            <li><a href="/contact.html">Contact us</a></li>
          </ul></li>
        <li class='has-sub'><a href='#'><span>Data</span></a>
          <ul>
            <li class='has-sub'><a href='#'><span>Human</span></a>
            <ul>
              <li><a href="/releases/current.html">Current Release</a></li>
              <li><a href="/releases/reference_releases.html">Reference Releases</a></li>
              <li><a href="/releases/">Release History</a></li>
              <li class="last"><a href="/releases/grch37_mapped_releases.html">GRCh37-mapped Releases</a></li>
            </ul></li>
            <li class='has-sub'><a href='#'><span>Mouse</span></a>
              <ul>
                <li><a href="/mouse_releases/current.html">Current Release</a></li>
                <li><a href="/mouse_releases/reference_releases.html">Reference Releases</a></li>
                <li class="last"><a href="/mouse_releases/">Release History</a></li>
              </ul></li>
            <li><a href="/data_format.html">Data Format</a></li>
            <li><a href="/gencode_biotypes.html">GENCODE Biotypes</a></li>
            <li class="last"><a href="/access_data.html">How to Access Data</a></li>
              </ul></li>
        <li class='has-sub'><a href='#'><span>Stats</span></a>
          <ul>
            <li class='has-sub'><a href='#'><span>Human</span></a>
              <ul>
                <li><a href="/stats/current.html">Current Release</a></li>
                <li><a href="/stats/reference_releases.html">Reference Releases</a></li>
                <li class="last"><a href="/stats/archive.html">Release History</a></li>
              </ul></li>
            <li class='has-sub'><a href='#'><span>Mouse</span></a>
            <ul>
                <li><a href="/mouse_stats/current.html">Current Release</a></li>
                <li><a href="/mouse_stats/reference_releases.html">Reference Releases</a></li>
                <li class="last"><a href="/mouse_stats/archive.html">Release History</a></li>
              </ul></li>
          </ul></li>
       <li class='has-sub'><a href='#'><span>Browser</span></a>
         <ul>
           <li><a href="/human_biodalliance.html">Human</a></li>
           <li><a href="/mouse_biodalliance.html">Mouse</a></li>
         </ul></li>
       <li class='last'><a href="http://gencodegenes.wordpress.com/" class="no-img">Blog</a></li>
      </ul>
    </div>

    <div id="content">
      
      
    <div class="panel">
      <h1>Release 19 (GRCh37.p13)</h1><br >
      <h3>GTF / GFF3 files</h3>
      <table class="filetable zebra-table" summary="List of GTF and GFF3 files in the Gencode release">
        <thead>
          <tr>
            <th style='width:20%'>
              Content
            </th>
            <th style='width:10%'>
              Regions
            </th>
            <th>
              Description
            </th>
            <th style='width:15%'>
              Download
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Comprehensive gene annotation
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>It contains the comprehensive gene annotation on the reference chromosomes only
                </li>
                <li>This is the <strong>main annotation file</strong> for most users
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.annotation.gtf.gz">GTF</a>
              &nbsp;&nbsp;<a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.annotation.gff3.gz">GFF3</a>
            </td>
          </tr>
          <tr>
            <td>
              Comprehensive gene annotation
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>It contains the comprehensive gene annotation on the reference chromosomes, scaffolds, assembly patches and
                alternate loci (haplotypes)
                </li>
                <li>This is a <strong>superset</strong> of the main annotation file
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.chr_patch_hapl_scaff.annotation.gtf.gz">GTF</a>
            </td>
          </tr>
          <tr>
            <td>
              Long non-coding RNA gene annotation
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>It contains the comprehensive gene annotation of lncRNA genes on the reference chromosomes
                </li>
                <li>This is a <strong>subset</strong> of the main annotation file
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.long_noncoding_RNAs.gtf.gz">GTF</a>
            </td>
          </tr>
          <tr>
            <td>
              PolyA feature annotation
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>It contains the polyA features (polyA_signal, polyA_site, pseudo_polyA) manually annotated by HAVANA on the
                reference chromosomes
                </li>
                <li>This dataset does <strong>not</strong> form part of the main annotation file
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.polyAs.gtf.gz">GTF</a>
            </td>
          </tr>
          <tr>
            <td>
              Consensus pseudogenes predicted by the Yale and UCSC pipelines
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>2-way consensus (retrotransposed) pseudogenes predicted by the Yale and UCSC pipelines, but not by HAVANA, on
                the reference chromosomes
                </li>
                <li>This dataset does <strong>not</strong> form part of the main annotation file
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.2wayconspseudos.gtf.gz">GTF</a>
            </td>
          </tr>
          <tr>
            <td>
              Predicted tRNA genes
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>tRNA genes predicted by ENSEMBL on the reference chromosomes using tRNAscan-SE
                </li>
                <li>This dataset does <strong>not</strong> form part of the main annotation file
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.tRNAs.gtf.gz">GTF</a>
            </td>
          </tr>
        </tbody>
      </table>
      <h3>Fasta files</h3>
      <table class="filetable zebra-table" summary="List of Fasta files in the Gencode release">
        <thead>
          <tr>
            <th style='width:20%'>
              Content
            </th>
            <th style='width:10%'>
              Regions
            </th>
            <th>
              Description
            </th>
            <th style='width:15%'>
              Download
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Protein-coding transcript sequences
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>Nucleotide sequences of coding transcripts on the reference chromosomes
                </li>
                <li>Transcript biotypes: protein_coding, nonsense_mediated_decay, non_stop_decay, IG_*_gene, TR_*_gene,
                polymorphic_pseudogene
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.pc_transcripts.fa.gz">Fasta</a>
            </td>
          </tr>
          <tr>
            <td>
              Protein-coding transcript translation sequences
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>Amino acid sequences of coding transcript translations on the reference chromosomes
                </li>
                <li>Transcript biotypes: protein_coding, nonsense_mediated_decay, non_stop_decay, IG_*_gene, TR_*_gene,
                polymorphic_pseudogene
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.pc_translations.fa.gz">Fasta</a>
            </td>
          </tr>
          <tr>
            <td>
              Long non-coding RNA transcript sequences
            </td>
            <td style='text-align:center'>
              CHR
            </td>
            <td>
              <ul>
                <li>Nucleotide sequences of long non-coding RNA transcripts on the reference chromosomes
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.lncRNA_transcripts.fa.gz">Fasta</a>
            </td>
          </tr>
          <tr>
            <td>
              Genome sequence (GRCh37.p13)
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Nucleotide sequence of the GRCh37.p13 genome assembly version on all regions, including reference
                chromosomes, scaffolds, assembly patches and haplotypes
                </li>
                <li>The sequence region names are the same as in the GTF/GFF3 files
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/GRCh37.p13.genome.fa.gz">Fasta</a>
            </td>
          </tr>
        </tbody>
      </table>
      <h3>Metadata files</h3>
      <table class="filetable zebra-table" summary="List of metadata files in the Gencode release">
        <thead>
          <tr>
            <th style='width:20%'>
              Content
            </th>
            <th style='width:10%'>
              Regions
            </th>
            <th>
              Description
            </th>
            <th style='width:15%'>
              Download
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Annotation remarks
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Remarks made during the manual annotation of the transcript
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.Annotation_remark.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              Exon annotation evidence
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Piece of evidence used in the annotation of an exon (usually peptides, mRNAs, ESTs)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.Exon_supporting_feature.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              Gene source
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Source of the gene annotation (Ensembl, Havana, Ensembl-Havana merged model or imported in the case of small
                RNA and mitochondrial genes)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.Gene_source.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              Gene symbol
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>HGNC approved gene symbol (from Ensembl xref pipeline)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.HGNC.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              PDB id
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>PDB entries associated to the transcript (from Ensembl xref pipeline)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.PDB.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              PolyA features
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Manually annotated polyA features overlapping the transcript 3'-end
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.PolyA_feature.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              PubMed id
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Pubmed ids of publications associated to the transcript (from HGNC website)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.Pubmed_id.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              RefSeq
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>RefSeq RNA and/or protein associated to the transcript (from Ensembl xref pipeline)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.RefSeq.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              Selenocysteine
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Amino acid position of a selenocysteine residue in the transcript
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.Selenocysteine.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              SwissProt
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>UniProtKB/SwissProt entry associated to the transcript (from Ensembl xref pipeline)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.SwissProt.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              Transcript source
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Source of the transcript annotation
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.Transcript_source.gz">Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              Transcript annotation evidence
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>Piece of evidence used in the annotation of the transcript
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href=
              "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.Transcript_supporting_feature.gz">
              Metadata</a>
            </td>
          </tr>
          <tr>
            <td>
              TrEMBL
            </td>
            <td style='text-align:center'>
              ALL
            </td>
            <td>
              <ul>
                <li>UniProtKB/TrEMBL entry associated to the transcript (from Ensembl xref pipeline)
                </li>
              </ul>
            </td>
            <td style='text-align:center'>
              <a href="ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_19/gencode.v19.metadata.TrEMBL.gz">Metadata</a>
            </td>
          </tr>
        </tbody>
      </table>
      <p>
        &nbsp; &nbsp; Statistics of this Gencode version are found <a href="/stats/archive.html#a19">here</a>.
      </p>
      <p>
        &nbsp; &nbsp; Format description of the GTFs is found <a href="/data_format.html">here</a>.
      </p>
      <p>
        &nbsp; &nbsp; More information about the GRCh37.p13 patches/scaffolds/haplotypes can be found <a href=
        "ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh37.p13">here</a>.
      </p>
    </div>
  
    </div>
    <div class="x">&nbsp;</div>
  </div>
  <div id="inst">
    <span id="tsandcs">
    <a href="/cookiespolicy.html">Cookies policy</a> | <a href="/s-inst/legal.html">Terms &amp; Conditions.</a>
    </span>
    <span id="hosttext">
    This site is hosted by the <a href="http://www.sanger.ac.uk/" class="no-img">Wellcome Trust Sanger Institute</a>.</span>
  </div>
   
  <noscript><div><img src="/piwik.php" alt="" style="width:1px;height:1px" ></div></noscript>
  <script type="text/javascript" src="/t/cssjs/MkhQQm0J9uPcikRWIbKXng.c.js"></script>
  <script type="text/javascript">
   $('#hnav>ul>li').hoverIntent(function(e){$(this).toggleClass('vis');});
  </script>
</body>
</html>
